builtin/providers/aws/resource_aws_db_event_subscription.go

```diff
@@ -0,0 +1,270 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDbEventSubscription() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDbEventSubscriptionCreate,
+		Read:   resourceAwsDbEventSubscriptionRead,
+		Update: resourceAwsDbEventSubscriptionUpdate,
+		Delete: resourceAwsDbEventSubscriptionDelete,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDbEventSubscriptionName,
+			},
+			"sns_topic": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"event_categories": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"source_ids": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+				// ValidateFunc: validateDbEventSubscriptionSourceIds,
+				// requires source_type to be set, does not seem to be a way to validate this
+			},
+			"source_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enabled": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsDbEventSubscriptionCreate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	name := d.Get("name").(string)
+	tags := tagsFromMapRDS(d.Get("tags").(map[string]interface{}))
+
+	sourceIdsSet := d.Get("source_ids").(*schema.Set)
+	sourceIds := make([]*string, sourceIdsSet.Len())
+	for i, sourceId := range sourceIdsSet.List() {
+		sourceIds[i] = aws.String(sourceId.(string))
+	}
+
+	eventCategoriesSet := d.Get("event_categories").(*schema.Set)
+	eventCategories := make([]*string, eventCategoriesSet.Len())
+	for i, eventCategory := range eventCategoriesSet.List() {
+		eventCategories[i] = aws.String(eventCategory.(string))
+	}
+
+	request := &rds.CreateEventSubscriptionInput{
+		SubscriptionName: aws.String(name),
+		SnsTopicArn:      aws.String(d.Get("sns_topic").(string)),
+		Enabled:          aws.Bool(d.Get("enabled").(bool)),
+		SourceIds:        sourceIds,
+		SourceType:       aws.String(d.Get("source_type").(string)),
+		EventCategories:  eventCategories,
+		Tags:             tags,
```

Comment:
Some fields here are optional, have you tried testing it when these fields are not set? Does it work as expected? Are you sure `d.GetOk` isn't necessary here?